### ISSUE TITLE
[feature request]: strip custom prefix from links

### ISSUE THREAD

**Issue by jalil-salame** [artempyanykh/marksman#376]
[Zola][https://getzola.org] (static site generator) uses a `[name](@/path/to/file.md)` syntax to resolve links (e.g. `[blog](@/blog/_index.md)` will resolve to `href="/blog/"` once compiled, and it points to the `blog/_index.md` file).

By stripping the `@/` prefix from the link (option in `config.toml` ideally) this could be supported, otherwise you just get a warning `Link to non-existent document '@/blog/_index.md'`.

My suggestion is to add a `strip_link_prefix`:tm: option to the configuration, this could be a single string or a list? (maybe someone needs this?), and before checking if the link is valid, this prefix would be stripped from it.

**Comment by artempyanykh** [comment]
Hi @jalil-salame! This sounds reasonable, but I'm guessing you'd still like other features like completion and rename refactor to put the `@/` prefix into the link, right?

**Comment by artempyanykh** [comment]
Fair enough! Although, I think that UX-wise the feature would feel somewhat incomplete without the support in completion and refactoring.

**Comment by rywng** [comment]
Hello, I'm also using zola and marksman, I am able to workaround this issue by doing the following steps: 

1. In your root directory, `ln -s content @`
2. `touch .marksman.toml`
3. in your markdown files, you should now complete the texts, but it is automatically parsed to something like `[text](/%40/_index.md)`
4. replace the `/%40` to `@`, I use a one-liner: `rg /%40 -l | xargs sed -i "s/\/%40/\@/"`
5. It should work now, I have tested goto definition, and the warning @jalil-salame mentioned is gone.
